standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -993,8 +993,37 @@ public boolean dropDatabase(String catName, String dbname)
 
   @Override
   public List<String> getDatabases(String catName, String pattern) throws MetaException {
+    catName = normalizeIdentifier(catName);
+    try {
+      return getDatabasesInternal(catName, pattern);
+    } catch (NoSuchObjectException e) {
+      throw new MetaException(ExceptionUtils.getStackTrace(e));
+    }
+  }
+
+  private List<String> getDatabasesInternal(String catName, String pattern)
+          throws MetaException, NoSuchObjectException {
+    return new GetHelper<List<String>>(catName, null, pattern, true, true) {
```

Comment:
I'd say two potential updates on this line.
First, I suspect this should be `new GetHelper<>(catname, null, null, true, true)` because GetHelper can unlikely handle a pattern. @dengzhhu653 We would like your advice on this hypothesis.
Second, if `GetHelper` accepts a pattern, this should be `GetHelper<>(catName, pattern, null, true, true)` because the given pattern is not that of tables but that of databases.